### DISCUSSION TITLE
YADC (yet another data contract)?

### DISCUSSION THREAD

**Discussion by dirkvandepoel** [datamesh-architecture/datacontract-specification#9] [General]
The datacontract.com website mentions it is inspired by other initiatives, what are the main reasons for not contributing to these other initiatives but creating your own data contract definition?

While it is still early days for data contracts, one of the main challenges is fragmentation  so curious whether it will converge towards some sort of a standard or everyone and every tool ends up creating their own.

**Comment by simonharrer** [comment]
Valid question. The Data Contract Specification follows these design principles:

- Feels familiar to people already used to OpenAPI and AsyncAPI
- Supports existing well-known formats to avoid unnecessary abstractions
- Supports contract-first and code-first approaches

And in these design principles, it differs a lot from the other proposed data contracts. As far as I know, it's the first one that follows OpenAPI/AsyncAPI conventions, allows bringing your own schema/quality checks, and with that, supports code-first approaches as well. This is in stark contrast what most other data contracts support.

We'll be talking with the other initiatives, whether we can converge to a more unified standard, still. But for that, we need to agree on common design principles.
